src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/TextDifferencing/SourceTextDiffer.CharDiffer.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.AspNetCore.Razor.TextDifferencing;
+
+internal partial class SourceTextDiffer
+{
+    private class CharDiffer : SourceTextDiffer
+    {
+        private readonly struct Buffer
+        {
+            public readonly char[] Array;
+            public readonly int Start;
+            public readonly int Length;
+
+            public Buffer(char[] array, int start, int length)
+                => (Array, Start, Length) = (array, start, length);
```

Comment:
neat, I knew about assignment like this but haven't seen it used often in constructors.